### DISCUSSION TITLE
Select input with nullable integer clears value entirely

### DISCUSSION THREAD

**Discussion by chillbram** [livewire/livewire#7455] [7. Bugs (archived)]
### Livewire version

v3.2.2

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Wirebox with the bug: https://wirebox.app/b/4kwoe

1. The form starts with a foo with value of `null`
2. Select Bar as value and click the button to see value set correctly to an integer
3. Now select None as value
4. Clicking the button will now throw an error (Typed property must not be accessed before initialization) because instead of being set to a `null` like value, the form property is completely unset.

### Additional information

This error does not happen with a `public ?string $foo` as an empty value will be transformed to an empty string. However, this is bug is still relevant for strings, as although no error will be thrown being set to an empty string is not the same as being set to the expected null.

I think being able to cast to nullable properties can be very valuable. I have several forms where fields should be nullable, but you always have to take into account that empty could mean `null` or `""`, which requires extra code in save functions for example. 

Maybe a solution is adding a `.nullable` modifier on wire:model, which on an empty value would cast to null?

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by DanielGSoftware** [comment]
Same problem here
